src/main/webapp/app/overview/course-lectures/course-lecture-details.component.ts

```diff
@@ -1,83 +1,67 @@
+    studentQuestions?: StudentQuestionsComponent;
+
     readonly LectureUnitType = LectureUnitType;
-    private studentQuestions?: StudentQuestionsComponent;
 
-    constructor(
-        private $location: Location,
-        private jhiWebsocketService: JhiWebsocketService,
-        private lectureService: LectureService,
-        private attachmentService: AttachmentService,
-        private authServerProvider: AuthServerProvider,
-        private route: ActivatedRoute,
-        private router: Router,
-        private fileService: FileService,
-    ) {
-        const navigation = this.router.getCurrentNavigation();
-        if (navigation && navigation.extras.state) {
-            const stateLecture = navigation.extras.state.lecture as Lecture;
-            if (stateLecture && stateLecture.startDate) {
-                stateLecture.startDate = moment(stateLecture.startDate);
-            }
-            if (stateLecture && stateLecture.endDate) {
-                stateLecture.endDate = moment(stateLecture.endDate);
+    constructor(private alertService: JhiAlertService, private lectureService: LectureService, private activatedRoute: ActivatedRoute, private fileService: FileService) {}
+
+    ngOnInit(): void {
+        this.activatedRoute.params.subscribe((params) => {
+            this.lectureId = +params['lectureId'];
+            if (this.lectureId) {
+                this.loadData();
             }
-            this.lecture = stateLecture;
-        }
+        });
     }
-    ngOnInit(): void {
-        this.subscription = this.route.params.subscribe((params) => {
-            if (!this.lecture || this.lecture.id !== params.lectureId) {
-                this.lecture = null;
-                this.lectureService.find(params.lectureId).subscribe((lectureResponse: HttpResponse<Lecture>) => {
-                    this.lecture = lectureResponse.body;
+
+    loadData() {
+        this.isLoading = true;
+        this.lectureService
+            .find(this.lectureId!)
+            .pipe(
+                finalize(() => {
+                    this.isLoading = false;
+                }),
+            )
+            .subscribe(
+                (findLectureResult) => {
+                    this.lecture = findLectureResult.body!;
                     if (this.lecture?.lectureUnits) {
                         this.lectureUnits = this.lecture.lectureUnits;
                     }
-                    if (this.studentQuestions && this.lecture) {
+                    if (this.studentQuestions) {
                         // We need to manually update the lecture property of the student questions component
                         this.studentQuestions.lecture = this.lecture;
                         this.studentQuestions.loadQuestions(); // reload the student questions
                     }
-                });
-            }
-        });
-    }
-
-    ngOnDestroy(): void {
-        if (this.subscription) {
-            this.subscription.unsubscribe();
```

Comment:
Why don't we need the ngOnDestroy anymore? Or rather the unsubscribe

Reply:
You do not need to unsubscribe for a parameters subscription. https://stackoverflow.com/questions/41138081/do-i-have-to-unsubscribe-from-activatedroute-e-g-params-observables